geode-core/src/main/java/org/apache/geode/cache/client/internal/pooling/ConnectionManagerImpl.java

```diff
@@ -338,6 +335,10 @@ public PooledConnection borrowConnection(ServerLocation server, long acquireTime
               }
             }
           }
+          if (null != connection) {
+            return connection;
+          }
+          throw new ServerConnectivityException(BORROW_CONN_ERROR_MSG + server);
```

Comment:
If the tryCreate() is true, and for any reason, unable to create connection, it should be in the loop till the specified timeout (server-timeout), right? throwing an exception here will return before the specified wait time.

Reply:
It is simular to borrowConnection() implementation, so we are reatempting, until we can try to create connection (until pool size drops bellow maximum). If server is unreachable, we will throw exception.

Reply:
OK, i get your point. I was thinking of the case, where server is busy when this connection is attempted. Also, do we need to handle GemFireSecurityException?

Reply:
As I said, logic is the same as for borrowConnection(), so I would not change this logic.